hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -478,21 +479,31 @@ async def async_cancel(self):
         ``True`` is returned if the job is cancelled. ``False`` is returned if
         the job has already completed.
         """
+        async def cancel_batch():
```

Comment:
I think we should just add `self.batch.cancel` to the finally block of the fetch_coro. Cancelling a completed batch is fine. That keeps everything simple: if the fetch_coro is cancelled, the BatchPoolFuture has been cancelled.